### ISSUE TITLE
Facing issue with custom treefile with systemd-tmpfiles failing to create directories under /var/spool

### ISSUE THREAD

**Issue by adityashah1212** [CentOS/sig-atomic-buildscripts#363]
Hi,
I am trying to create a custom tree for a specific use case (Purposely being vague about the use case). The tree creation goes fine, the tree is created fine and the tree is created. Now when I try to pxe into this live environment, and kickstart, at ostreesetup stage stage, it says Deployment complete but fails after that saying 

![kickstart_failure](https://user-images.githubusercontent.com/25403870/48003576-d1145e00-e0dc-11e8-9f9a-39a1a1e6a406.png)

I tried running that same command, but it seems that it is unable to find the postfix user, though postfix user is present under /mnt/sysimage/ostree/deploy/\<build-name\>/deploy/\<Not typing the huge hash\>/usr/lib/passwd. So I am not sure what might be going wrong. 

I am using the squashfs.img and kernel from the http://cloud.centos.org/centos/7/atomic/images/CentOS-Atomic-Host-7.1808-Installer.iso

BTW, I have applied https://github.com/rhinstaller/anaconda/pull/4/ since I am using a self signed https server for my testing. And patch might still remain in use for sometime.

I would appreciate any help you guys can offer.

**Comment by jlebon** [comment]
Hmm, my guess here is that `systemd-tmpfiles` doesn't actually make use of `/usr/lib/{passwd,group}` in the deployment root since it's not going through nss-altfiles? In which case, hmm yeah that's a tricky problem. I'm impressed we've avoided this issue so far.

One possible way to hack around this is in your treecompose postprocess script to scrub out that tmpfiles entry; it's likely postfix will just create it itself on startup. If it gives you errors, you can also try subbing out the username in the tmpfiles entry for the postfix uid.

**Comment by adityashah1212** [comment]
I am kind of new to this so sorry for not understanding, but could you please elaborate on what do you mean by
>  treecompose postprocess script to scrub out that tmpfiles entry

and

> you can also try subbing out the username in the tmpfiles entry for the postfix uid

**Comment by jlebon** [comment]
It's a script that runs after packages are installed, but before the OSTree commit is written. See `postprocess-script` in https://rpm-ostree.readthedocs.io/en/latest/manual/treefile/. An example is in this repo itself: https://github.com/CentOS/sig-atomic-buildscripts/blob/c1df093436f69038bf2fe773ebdc2dc8359384ea/treecompose-post.sh.

**Comment by adityashah1212** [comment]
Ah I knew about the script, I was confused about the changes you were asking me to do, but I think I just figured out what you meant. Let me give a try.

Thanks :)

**Comment by adityashah1212** [comment]
Ok, the offending entry is a part of /usr/lib/tmpfiles.d/rpm-ostree-1-autovar.conf and this file is not present during treecompose-post.sh execution. Hence I tried to copy the usr/lib/passwd to etc/passwd under the ostree deployed directory, but that doesn't fix this error. The only way I found to fix this was to copy the passwd and group files to /etc of the livefs before running that systemd-tmpfiles command mannually. But I don't know if that is the right approach. I tried running this in chroot, but that is causing even more problems as it is not able to resolve the machine id and fails a creating the directory related for journal.
